cmd/injector/main.go

```diff
@@ -17,95 +16,165 @@ import (
+	// This is provided by environment variables.
+	type Config struct {
+		CertFile            string `env:"DSV_CERT"  envDefault:"${HOME}/tls/cert.pem" envExpand:"true"`                       // Cert is the path to the public certificate file in PEM format.
+		KeyFile             string `env:"DSV_KEY" envDefault:"${HOME}/tls/key.pem" envExpand:"true"`                          // Key is the path to the private key file in PEM format.
+		CredentialsJSONFile string `env:"DSV_CREDENTIALS_JSON" envDefault:"${HOME}/credentials/config.json" envExpand:"true"` // CredentialsJSONFile is the path to the JSON formatted credentials file that is mounted as a secret.
+		ServerAddress       string `env:"DSV_SERVER_ADDRESS" envDefault:":18543"`                                             // ServerAddress is the address to listen on, e.g., 'localhost:8080' or ':8443'
+		Debug               bool   `env:"DSV_DEBUG" envDefault:"false"`                                                       // Debug enables debug logging.
+	}
 
-	server := new(http.Server)
+	cfg := Config{}
+	err := env.Parse(&cfg)
+	if err != nil {
+		log.Error().Err(err).Msg("unable to parse environment variables")
+	}
+	if cfg.Debug {
+		logger.EnableDebug()
+		log.Info().Msg("debug logging enabled")
+	}
+	log.Info().Strs("args", args).Msg("starting syncer, args passed, but not used, as environment variables are used instead")
 
-	flag.StringVar(&server.Addr, "address", ":18543", "the address to listen on, e.g., 'localhost:8080' or ':8443'")
-	flag.Parse()
+	log.Info().
+		Str("version", version).
+		Str("commit", commit).
+		Str("date", date).
+		Msg("syncer version information")
 
-	credentials, err := config.GetCredentials(credentialsFile)
+	credentials, err := config.GetCredentials(cfg.CredentialsJSONFile)
 	if err != nil {
-		log.Fatalf("unable to process credentials file '%s': %s", credentialsFile, err)
+		log.Error().Err(err).Str("credential-json", cfg.CredentialsJSONFile).Msg("unable to process credentials file")
+		return fmt.Errorf("unable to process credentials file %q: %w", cfg.CredentialsJSONFile, err)
 	}
-	log.Printf("[INFO] success loading %d credential sets: [%s] from '%s'",
-		len(*credentials),
-		strings.Join(credentials.Names(), ", "),
-		credentialsFile,
-	)
-
-	if cert, err := tls.LoadX509KeyPair(certFile, keyFile); err == nil {
-		server.TLSConfig = &tls.Config{Certificates: []tls.Certificate{cert}}
+	log.Info().
+		Str("credential_names", strings.Join(credentials.Names(), ", ")).
+		Str("credential_file", cfg.CredentialsJSONFile).
+		Msg("credentials loaded from JSON file")
+	var tlsConfig *tls.Config
+	if cert, err := tls.LoadX509KeyPair(cfg.CertFile, cfg.KeyFile); err == nil {
+		tlsConfig = &tls.Config{Certificates: []tls.Certificate{cert}}
+		log.Info().Str("cert", cfg.CertFile).Str("key", cfg.KeyFile).Msg("LoadX509KeyPair")
 	} else {
-		log.Fatalf("unable to load keypair for TLS: %s", err)
+		log.Error().Err(err).Msgf("unable to load keypair for TLS: %s", err)
```

Comment:
In reviewing the chart it mounts this so while it might not working for a local run outside k8s (which would require havin g local signed cert generated), I'm going to opt for simplicity right now and require this and can revert if needed later or improve for local running outside k8s. I don't do that, and th eonly directions on doing that are from the prior maintainers so it's not a priority to maintain that right now.